### DISCUSSION TITLE
ElectriQ Airflex 15W Dashboard behaviour

### DISCUSSION THREAD

**Discussion by Pel1can111** [make-all/tuya-local#609] [Q&A]
I was curious if there were any other owners of this portable aircon unit that have had much luck with home assistant integration?

Firstly it seems this integration offers the best experience for me. The standard Tuya integration had issues with the reported temp and would also not control the unit properly.

This one seems to work but the only issue I'm having is that selecting modes (heat,cool,fan,dry) from the climate dashboard does not work properly. Opening the settings for the unit and selecting the mode from the "operation" drop down menu also fails but selecting the same option from the "preset" drop down menu works perfectly. Is anyone else experiencing this?

Thanks

**Comment by matt7aylor** [comment]
I am (or was) experiencing the same behaviour as you, I needed to use the Preset option to change the mode rather than the Operation selector, the only thing that did work from the operation selection was on/off. However on updating the plugin I now do not have the preset selector any more so will have to dig into this a little further before I will be able to change modes again.

**Comment by matt7aylor** [comment]
Reverting the change to [custom_components/tuya_local/devices/electriq_airflex15w_heatpump.yaml](https://github.com/make-all/tuya-local/commit/acce76acdcea31d6931f4cb84fe27f0e1a3c3645#diff-7112db12b34ce7ed37bec22ff3f9521013b9f8b3f157b974f1ff96d21bdea7ac) in acce76a does restore the ability to change mode using the "Preset" selector for me, but I assume it is meant to be happening automatically when changing the operation and that is the underlying issue that results in the odd behaviour we are seeing.

**Reply by make-all** [reply]
It is possible that this device does not handle two dps being sent at the same time (the hvac_mode selector is combining both power and mode, so always sends both at the same time to the device)

**Reply by Pel1can111** [reply]
Thanks for the replies, its nice to see I'm not the only one with issues. I'll have a look at reverting that change as since then I can only turn the unit on and off (not a huge problem as I got the unit for cooling).

**Reply by matt7aylor** [reply]
> It is possible that this device does not handle two dps being sent at the same time (the hvac_mode selector is combining both power and mode, so always sends both at the same time to the device)

That sounds like a good theory, I had a little play with the yaml earlier to see if I could come up with something that would do both on/off and mode together but I'm not at all familiar with it so didn't get anywhere that wasn't more likely to be a limitation with my implementation. Do you have something you could suggest as a simple test for this and I'd be happy to give it a try.

**Comment by Pel1can111** [comment]
Seems the latest upgrade has prevented home assistant from turning the air conditioner on/off. Manually turning it on results in the ability to change presets working again but this, unfortunately, is a worse condition than before as I rarely changed the preset

**Comment by make-all** [comment]
I have updated the config with the following changes to allow changing the power and mode separately.

1. restored the preset_mode in parallel with hvac_mode
2. added a separate switch entity for the power.

The latter is not strictly necessary, as the climate.toggle service call should do the same thing, but it makes the UI easier as I don't think toggle is exposed anywhere other than automations.

**Reply by Pel1can111** [reply]
I have added the changes and that seems to do the trick. 

Power button works as expect and turns the unit on and off
Preset_mode changes the device mode as expected now 

Thanks for getting this back up and running with HA given the current heat wave! 👍
